python/mxnet/optimizer.py

```diff
@@ -299,8 +300,13 @@ def update(self, index, weight, grad, state):
         if state:
             mom = state
             mom[:] *= self.momentum
-            mom[:] += -lr * (grad + wd * weight)
-            weight[:] += mom
+            grad += wd * weight
```

Comment:
What problem can it cause?
Even though it could cause problem writing to the original gradient, 
in this previous line we already done this
`grad = grad * self.rescale_grad`, 
So the grad here is no longer the original grad passed to this function.